lib/events/dynamoevents/dynamoevents.go

```diff
@@ -499,45 +503,66 @@ func (l *Log) SearchEvents(fromUTC, toUTC time.Time, filter string, limit int) (
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
-	input := dynamodb.QueryInput{
-		KeyConditionExpression:    aws.String(query),
-		TableName:                 aws.String(l.Tablename),
-		ExpressionAttributeValues: attributeValues,
-		IndexName:                 aws.String(indexTimeSearch),
-	}
-	start := time.Now()
-	out, err := l.svc.Query(&input)
-	if err != nil {
-		return nil, trace.Wrap(err)
-	}
-	g.WithFields(log.Fields{"duration": time.Since(start), "items": len(out.Items)}).Debugf("Query completed.")
+
+	var lastEvaluatedKey map[string]*dynamodb.AttributeValue
 	var total int
-	for _, item := range out.Items {
-		var e event
-		if err := dynamodbattribute.UnmarshalMap(item, &e); err != nil {
-			return nil, trace.BadParameter("failed to unmarshal event for %v", err)
+
+	// We iterate over the responses until we roughly hit ~90MB in size.
+	// This is an arbitrary limit to prevent runaway loops.
+	const maxResponseSizeKbs = maxPageSizeKbs * 100
+
+	for responseSize := 0; responseSize < maxResponseSizeKbs; responseSize += maxPageSizeKbs {
```

Comment:
Sorry, I may be missing something but this condition is pretty confusing to me. The idea of limiting by memory makes sense, however I don't see how it's actually enforced here - i.e. the `responseSize` is not the actual size of the response but basically just a "weighted" iteration number.

So how is this different from, say, just `for i := 0; i < 100; i += 1`? If the intent is to convey the fact that limiting number of iterations to a 100 will limit max returned response size to 90MB, then why not just put that in the comment?

Reply:
You are right as we are not counting the actual size it makes sense to use a basic loop. I've addressed the changes.